File: StykerDG.FarmForge.DataModel/Models/CropType.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace StrykerDG.FarmForge.DataModel.Models
{
    [Table("CropType")]
    public class CropType : BaseModel
    {
        [Key]
        public int CropTypeId { get; set; }
        public int CropClassificationId { get; set; }
        [ForeignKey("CropClassificationId")]
        public CropClassification Classification { get; set; }
        public string Name { get; set; }
        public string Label { get; set; }
        public long? AverageGermination { get; set; }
        public long? AverageTimeToHarvest { get; set; }
        public double? AverageYield { get; set; }
        public ICollection<CropVariety> Varieties { get; set; }
    }
}

File: StrykerDG.FarmForge.Client/farmforge_client/lib/models/crops/crop_type.dart
import 'package:farmforge_client/models/crops/crop_classification.dart';
import 'package:farmforge_client/models/crops/crop_variety.dart';

class CropType {
  int cropTypeId;
  int cropClassificationId;
  CropClassification cropClassification;
  String name;
  String label;
  int averageGermination;
  int averageTimeToHarvest;
  double averageYield;
  List<CropVariety> varieties;

  CropType({
    this.cropTypeId,
    this.cropClassificationId,
    this.cropClassification,
    this.name,
    this.label,
    this.averageGermination,
    this.averageTimeToHarvest,
    this.averageYield,
    this.varieties
  });

  factory CropType.fromMap(Map<String, dynamic> data) {
    List<CropVariety> cropVarieties = new List<CropVariety>();
    if(data['Varieties'] != null && data['Varieties'].length > 0)
      data['Varieties'].forEach((varietyData) {
        CropVariety variety = CropVariety.fromMap(varietyData);
        cropVarieties.add(variety);
      });

    CropClassification classification = data['CropClassification'] != null
      ? CropClassification.fromMap(data['CropClassification'])
      : null;

    return CropType(
      cropTypeId: data['CropTypeId'],
      cropClassificationId: data['CropClassificationId'],
      cropClassification: classification,
      name: data['Name'],
      label: data['Label'],
      averageGermination: data['AverageGermination'],
      averageTimeToHarvest: data['AverageTimeToHarvest'],
      averageYield: data['AverageYield'],
      varieties: cropVarieties
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'CropTypeId': this.cropTypeId,
      'CropClassificationId': this.cropClassificationId,
      'CropClassification': null,
      'Name': this.name,
      'Label': this.label,
      'AverageGermination': this.averageGermination,
      'AverageTimeToHarvest': this.averageTimeToHarvest,
      'AverageYield': this.averageYield,
      'Varieties': null
    };
  }
}